articles/quickstart/native/ios-swift-siwa/_includes/_login_centralized.md

```diff
@@ -0,0 +1,306 @@
+:::
+
+At this point, you will have a few compile errors to deal with. Before moving forward with all the code, let's remove those errors by providing some basic delegates.
+
+### Add authorization extensions
+
+Add the following to the end of the `ViewController.swift` file. This enables the authorization flow UI to be displayed from our controller:
+
+```swift
+extension ViewController: ASAuthorizationControllerPresentationContextProviding {
+    func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        return self.view.window!
+    }
+}
+```
+
+Next, add a stub delegate for the `ASAuthorizationController` to the end of the `ViewController.swift` file. This code will handle the response sent to the `ASAuthorizationController`. You will come back to this one later.
+
+```swift
+extension ViewController: ASAuthorizationControllerDelegate {
+}
+```
+
+### Add the Sign In With Apple capability
+
+To enable Sign In With Apple in your application, you need to enable this capability to your app, as shown:
+
+![Add capabilities](/media/articles/ios/swift/add-capability.png)
+
+:::panel Checkpoint
+Start the application and click the **Sign In With Apple** button. You should see the default sign-in dialog appear:
+
+![Default sign-in dialog](/media/articles/ios/swift/authz-dialog.png)
+:::
+
+## Process the Authorization Response
+
+Add the following to the end of your `ViewController.swift` file.
+
+```swift
+extension ViewController: ASAuthorizationControllerDelegate {
+    // Handle authorization success
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+
+        if let appleIDCredential = authorization.credential as? ASAuthorizationAppleIDCredential {
+            // Success
+        }
+    }
+
+    // Handle authorization failure
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: Error) {
+        print("Authorization Failed: \(error)")
+    }
+}
+```
+
+### Inspect the response
+
+Upon successful authorization, you will be able to access the following information:
+
+- UUID, team-scoped user ID
+- Verification Data
+  - ID Token, Authorization Code
+- Account Information
+  - Name, Verified Email Address (Original or relay address)
+- Real user indicator
+  - High confidence indicator that the user is who they say they are
+
+:::note
+The Account Information is ONLY sent on the first successful authorization
+:::
+
+## Auth0 Token Exchange
+
+Now that you have a successful authorization response, you can use the `authorizationCode` information to perform a token exchange to gain access to Auth0 credentials, such as the ID and Access Tokens, as in the following example:
```

Comment:
Now that you have a successful authorization response, you can use the `authorizationCode` to perform a token exchange. The token exchange will give you access to Auth0 credentials, such as the ID and Access Tokens. For example: